docs/release-notes/8.4.2-2.md

```diff
@@ -0,0 +1,84 @@
+# Percona Server for MySQL 8.4.2-2 (2024-10-31)
+
+Get started with [Quickstart Guide for Percona Server for MySQL].
+
+[Percona Server for MySQL 8.4.2-2] includes all the features and bug fixes available in the
+MySQL 8.4 Community Edition in addition to enterprise-grade features developed by Percona.
+
+## Release highlights
+
+Improvements and bug fixes introduced by Oracle for MySQL 8.4.1 and 8.4.2 and included in Percona Server for MySQL are the following:
+
+* MySQL stopped unexpectedly during an UPDATE after an ALTER TABLE operation.
+
+* Shutting down the server after an XA START with an empty XA transaction caused it to stop unexpectedly.
+
+* Shutting down the replication applier or binlog applier during an empty XA transaction caused the system to stop unexpectedly.
+
+* The result from a spatial index with a column containing a spatial reference identifier (SRID) was empty. Using FORCE INDEX to scan this index caused an assertion error.
+
+* In some cases, after creating more than 8000 tables, the server failed to restart. 
+
+* Startup tablespace file scanning performance was improved.
+
+Find the complete list of bug fixes and changes in the [MySQL 8.4.1 Release Notes] and [MySQL 8.4.2 Release Notes].
+
+## Bug fixes
+
+* [PS-8057]: `slow_query_log_file` does not match the filename defined in my.cnf.
+
+* [PS-9144]: Missing rows after running a null `ALTER` with `ALGORTITHM=INPLACE`.
+
+* [PS-9214]: An `ALTER` table online results in a "duplicate key" error on the primary key (only index).
+
+* [PS-9306]: The following MySQL versions unexpectedly exit if the database has more than 10K tables:
+
+    * 8.0.38
+    
+    * 8.4.1
+
+    * 9.0.0
+
+* [PS-9314]: Using a JSON_TABLE in Percona Server for MySQL 8.0.36 causes a signal 11 error. 
+
+* [PS-9379]: On some platforms, using "-Iatomic" caused compilation errors when the compiler had built-in "atomic" support.
```

Comment:
PS-9379 is too internal - it was caught before this was released and fixed immediately